gns3server/handlers/api/controller/node_handler.py

```diff
@@ -59,16 +61,40 @@ class NodeHandler:
     @Route.get(
         r"/projects/{project_id}/nodes",
         parameters={
-            "project_id": "Project UUID"
+            "project_id": "Project UUID",
+            "node_type": "Node type",
+            "status": "Status",
+            "name": "Node name",
+            "locked": "Locked"
```

Comment:
I think this is the cause of the `AssertionError: assert 500 == 200` below. The parameters dict is not for optional queries (I know the name is misleading).